blog/en/blog/2023/12/14/apisix-plugins-priority-leaky-abstraction.md

```diff
@@ -0,0 +1,985 @@
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>tcp-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>kafka-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>rocketmq-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>udp-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>clickhouse-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>syslog</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>log-rotate</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <td><code>error-log-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <td><code>sls-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>google-cloud-logging</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>splunk-hec-logging</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>file-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>loggly</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>elasticsearch-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>tencent-cloud-cls</code></td>
+  <td></td>
+  <td>X</td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+  <td>X</td>
+</tr>
+<tr>
+  <td><code>loki-logger</code></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td>X</td>
+  <td>X</td>
+</tr>
+</tbody>
+<thead>
+<tr>
+  <th><em>Other</em></th>
+  <th><code>rewrite</code></th>
+  <th><code>access</code></th>
+  <th><code>before_proxy</code></th>
+  <th><code>header_filter</code></th>
+  <th><code>body_filter</code></th>
+  <th><code>log</code></th>
+</tr>
+</thead>
+<tbody>
+<tr>
+  <td><code>serverless</code></td>
+  <td></td>
+  <td>X</td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <td><code>openwhisk</code></td>
+  <td></td>
+  <td>X</td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <td><code>dubbo-proxy</code></td>
+  <td></td>
+  <td>X</td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <td><code>kafka-proxy</code></td>
+  <td></td>
+  <td>X</td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+</tbody>
+</table>
```

Comment:
Could you provide a screenshot on this part?

Reply:
<img width="847" alt="Screenshot 2023-12-11 at 08 12 24" src="https://github.com/apache/apisix-website/assets/752258/5c4dacf7-cfdd-4273-9fd8-ef26cbae1017">

Reply:
OK LGTM